src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedFieldSymbolBase.cs

```diff
@@ -56,6 +56,15 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
             {
                 AddSynthesizedAttribute(ref attributes, compilation.SynthesizeDynamicAttribute(this.Type, this.CustomModifiers.Length));
             }
+
+            if (Type.ContainsTuple() &&
+                compilation.HasTupleNamesAttributes &&
+                compilation.CanEmitBoolean() &&
+                compilation.CanEmitString())
+            {
+                AddSynthesizedAttribute(ref attributes,
+                    compilation.SynthesizeTupleNamesAttribute(Type));
+            }
```

Comment:
Do we similar handling in `SynthesizedParameterSymbol` and `SynthesizedImplementationMethod`?